controllers/vspherecluster_controller.go

```diff
@@ -301,14 +301,12 @@ func (r clusterReconciler) reconcileNormal(ctx *context.ClusterContext) (reconci
 		return reconcile.Result{}, err
 	}
 
-	if cloudProviderConfigurationAvailable(ctx) {
-		if err := r.reconcileVCenterConnectivity(ctx); err != nil {
-			conditions.MarkFalse(ctx.VSphereCluster, infrav1.VCenterAvailableCondition, infrav1.VCenterUnreachableReason, clusterv1.ConditionSeverityError, err.Error())
-			return reconcile.Result{}, errors.Wrapf(err,
-				"unexpected error while probing vcenter for %s", ctx)
-		}
-		conditions.MarkTrue(ctx.VSphereCluster, infrav1.VCenterAvailableCondition)
```

Comment:
This will break since the method `reconcileVCenterConnectivity` depends on the Datacenter value from the cloud provider configuration. Also, this change is included in another PR raised to remove the deprecated types in v1alpha4. I would suggest let us wait for that to merge since it removes all the deprecated types together, and alters the controller logic accordingly.

Reply:
We also saw that `reconcileVCenterConnectivity` refers to the cloud provider's Datacenter. We were under the impression that the Datacenter would be empty string in the case the cloud provider configuration was not present. The reachability and connectivity to the VCenter can still be verified in that case.

Reply:
Just skimmed #1095 and agree it would fix #1209, so happy to see it resolved that way if it gets merged first. Certainly don't want to get in the way of merging that.

If valuable, we could slim this PR down to just add  the test coverage included here. 

We were similarly under the impression initially that the `cloudProviderConfigurationAvailable` guard was needed given that `reconcileVCenterConnectivity` accesses `Datacenter`, but on closer inspection, decided it was safe to remove that guard as @tylerschultz mentioned for the reasons he stated above.

Given that, we are also considering opening a followup PR against the `release-0.7` with this more targeted fix if that would be welcome.

Reply:
> Given that, we are also considering opening a followup PR against the release-0.7 with this more targeted fix if that would be welcome.

Here's that other PR: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1216